release v1.1.0-rc.1 (#2072)
pingcap/tidb-operator: CHANGELOG-1.1.md

# TiDB Operator v1.1.0-rc.1 Release Notes

This is a release candidate of `v1.1.0`, which focuses on the usability, extensibility and security of TiDB Operator. While we encourage usage in non-critical environments, it is **NOT** recommended to use this version in critical environments.

## Action Required

> - Add field `tlsClient` in the backup and restore spec to control whether to enable TLS for connection with TiDB Server and also support specifying a secret name that including the cert. ([#2003](https://github.com/pingcap/tidb-operator/pull/2003), [@shuijing198799](https://github.com/shuijing198799))

Reviewer: I suggest not using the short version of words, such as `spec` and `cert`
```suggestion
- Add field `tlsClient` in the backup and restore spec, which controls whether to enable TLS for connection with TiDB Server and support specifying a secret name that includes the cert. ([#2003](https://github.com/pingcap/tidb-operator/pull/2003), [@shuijing198799](https://github.com/shuijing198799))
```

- change the field `BRConfig.EnableTLSClient` as type `bool` to `BRConfig.TLSCluster` as type `*TLSCluster` as the TLS trigger. ([#1988](https://github.com/pingcap/tidb-operator/pull/1988), [@shuijing198799](https://github.com/shuijing198799))

> 

Reviewer 2: And `BRConfig.TLSCluster` is removed in #2062, now backup controller retrieves the tlsCluster and tlsClient config from tc and passes them to the backup-manager, we can add one item to the `Notable Changes` for BR TLS support.

> 

Author: it makes sense.

## Other Notable Changes

- Fix the PD `location-labels` configuration ([#1941](https://github.com/pingcap/tidb-operator/pull/1941), [@aylei](https://github.com/aylei))
- Able to pause and unpause tidb cluster deployment via `spec.paused`. ([#2013](https://github.com/pingcap/tidb-operator/pull/2013), [@cofyc](https://github.com/cofyc))

> - Default the `max-size` and `max-backups` for tidb server configuration with `300` and `3` if the tidb cluster was deployed by CR. ([#2045](https://github.com/pingcap/tidb-operator/pull/2045), [@Yisaer](https://github.com/Yisaer))

Reviewer: ```suggestion
- Default the `max-size` and `max-backups` for tidb server configuration to `300` and `3` if the tidb cluster was deployed by CR. ([#2045](https://github.com/pingcap/tidb-operator/pull/2045), [@Yisaer](https://github.com/Yisaer))
```

> - Able to configure custom env for components. ([#2052](https://github.com/pingcap/tidb-operator/pull/2052), [@cofyc](https://github.com/cofyc))

Reviewer: Please use `environment` rather than `env` to avoid confusion

- Fix the error that `kubectl get tc` cannot show correct images ([#2031](https://github.com/pingcap/tidb-operator/pull/2031), [@Yisaer](https://github.com/Yisaer))

> - 1. Default the `spec.tikv.maxfailureOver` and `spec.tidb.maxfailureOver` to 3 when they are no defined

Reviewer: ```suggestion
  1. Default the `spec.tikv.maxfailureOver` and `spec.tidb.maxfailureOver` to 3 when they are not defined
```

>   2. Change the action that when `maxfailureOver` is zero, operator would not failover instead of creating limitless instances. ([#2015](https://github.com/pingcap/tidb-operator/pull/2015), [@Yisaer](https://github.com/Yisaer))

Reviewer: @Yisaer This sentence is confusing. Could you pls explain?

Author: I'm going to simplify it to "Disable auto-failover when `maxFailoverCount` is set to `0`".

- Support deploying TiDB Cluster with TidbCluster and TidbMonitor CRs via Terraform on ACK. ([#2012](https://github.com/pingcap/tidb-operator/pull/2012), [@DanielZhangQD](https://github.com/DanielZhangQD))

> - Update PDConfig for TidbCluster to pd v3.1.0 ([#1928](https://github.com/pingcap/tidb-operator/pull/1928), [@Yisaer](https://github.com/Yisaer))

Reviewer: ```suggestion
- Update PDConfig for TidbCluster to PD v3.1.0 ([#1928](https://github.com/pingcap/tidb-operator/pull/1928), [@Yisaer](https://github.com/Yisaer))
```

- Support deploying TiDB Cluster with TidbCluster and TidbMonitor CRs via Terraform on AWS. ([#2004](https://github.com/pingcap/tidb-operator/pull/2004), [@DanielZhangQD](https://github.com/DanielZhangQD))

> - Update TidbConfig for TidbCluster to tidb v3.1.0 ([#1906](https://github.com/pingcap/tidb-operator/pull/1906), [@Yisaer](https://github.com/Yisaer))

Reviewer: ```suggestion
- Update TidbConfig for TidbCluster to TiDB v3.1.0 ([#1906](https://github.com/pingcap/tidb-operator/pull/1906), [@Yisaer](https://github.com/Yisaer))
```

- Allow users to define resources for initContainers in TiDB initializer job. ([#1938](https://github.com/pingcap/tidb-operator/pull/1938), [@tfulcrand](https://github.com/tfulcrand))
- TLS support for Pump and Drainer ([#1979](https://github.com/pingcap/tidb-operator/pull/1979), [@weekface](https://github.com/weekface))
- Add doc and examples for auto-scaler and intializer ([#1772](https://github.com/pingcap/tidb-operator/pull/1772), [@Yisaer](https://github.com/Yisaer))

> - 1. Add check to guaranteeing the nodePort won't be changed if the serviceType of TidbMonitor is Nodeport

Reviewer 2: ```suggestion
- 1. Add check to guarantee the NodePort won't be changed if the serviceType of TidbMonitor is NodePort
```

>   2. Add EnvVar sort to avoid the monitor controller render different result from the same TidbMonitor spec.

Reviewer 2: ```suggestion
  2. Add EnvVar sort to avoid the monitor controller rendering different results from the same TidbMonitor spec.
```

>   3. Fix the problem that Tidbmonitor LoadBalance IP is not used. ([#1962](https://github.com/pingcap/tidb-operator/pull/1962), [@Yisaer](https://github.com/Yisaer))

Reviewer: ```suggestion
  3. Fix the problem that Tidbmonitor LoadBalancer IP is not used. ([#1962](https://github.com/pingcap/tidb-operator/pull/1962), [@Yisaer](https://github.com/Yisaer))
```

# TiDB Operator v1.1.0-rc.1 Release Notes

This is a release candidate of `v1.1.0`, which focuses on the usability, extensibility and security of TiDB Operator. While we encourage usage in non-critical environments, it is **NOT** recommended to use this version in critical environments.

## Action Required

> 

Author: I'll remove all periods.

> 

Author: done

- `--advertise-address` will be configured for `tidb-server` which will trigger rolling-upgrade for `tidb-server` component, you can set `spec.paused` to `true` to avoid it before upgrading tidb-operator and set it back to `false` when it's ready to upgrade your tidb server. ([#2076](https://github.com/pingcap/tidb-operator/pull/2076), [@cofyc](https://github.com/cofyc))

> - Add field `tlsClient` in the backup and restore spec, which controls whether to enable TLS for connection with TiDB Server and support specifying a secret name that includes the cert. ([#2003](https://github.com/pingcap/tidb-operator/pull/2003), [@shuijing198799](https://github.com/shuijing198799))

Reviewer 2: ```suggestion
- Add field `tlsClient.tlsSecret` in the backup and restore spec, which supports specifying a secret name that includes the cert. ([#2003](https://github.com/pingcap/tidb-operator/pull/2003), [@shuijing198799](https://github.com/shuijing198799))
```

Reviewer 2: Whether to enable TLS is retrieved from tc.

Author: updated

## Other Notable Changes

- Use `tidb-lightning` in `Restore` instead of `loader`. ([#2068](https://github.com/pingcap/tidb-operator/pull/2068), [@Yisaer](https://github.com/Yisaer))
- Add `cert-allowed-cn` support to TiDB components ([#2061](https://github.com/pingcap/tidb-operator/pull/2061), [@weekface](https://github.com/weekface))
- Fix the PD `location-labels` configuration ([#1941](https://github.com/pingcap/tidb-operator/pull/1941), [@aylei](https://github.com/aylei))
- Able to pause and unpause tidb cluster deployment via `spec.paused`. ([#2013](https://github.com/pingcap/tidb-operator/pull/2013), [@cofyc](https://github.com/cofyc))

> - Default the `max-backups` for tidb server configuration to `3` if the tidb cluster was deployed by CR. ([#2045](https://github.com/pingcap/tidb-operator/pull/2045), [@Yisaer](https://github.com/Yisaer))

Reviewer: ```suggestion
- Default the `max-backups` for tidb server configuration to `3` if the TiDB cluster is deployed by CR. ([#2045](https://github.com/pingcap/tidb-operator/pull/2045), [@Yisaer](https://github.com/Yisaer))
```

Author: updated

- Able to configure custom environments for components. ([#2052](https://github.com/pingcap/tidb-operator/pull/2052), [@cofyc](https://github.com/cofyc))
- Fix the error that `kubectl get tc` cannot show correct images ([#2031](https://github.com/pingcap/tidb-operator/pull/2031), [@Yisaer](https://github.com/Yisaer))

> - 1. Default the `spec.tikv.maxFailoverCount` and `spec.tidb.maxFailoverCount` to 3 when they are not defined

Reviewer: ```suggestion
- 1. Default the `spec.tikv.maxFailoverCount` and `spec.tidb.maxFailoverCount` to `3` when they are not defined
```

Author: updated

  2. Disable auto-failover when `maxFailoverCount` is set to `0` ([#2015](https://github.com/pingcap/tidb-operator/pull/2015), [@Yisaer](https://github.com/Yisaer))
- Support deploying TiDB Cluster with TidbCluster and TidbMonitor CRs via Terraform on ACK. ([#2012](https://github.com/pingcap/tidb-operator/pull/2012), [@DanielZhangQD](https://github.com/DanielZhangQD))
- Update PDConfig for TidbCluster to PD v3.1.0 ([#1928](https://github.com/pingcap/tidb-operator/pull/1928), [@Yisaer](https://github.com/Yisaer))
- Support deploying TiDB Cluster with TidbCluster and TidbMonitor CRs via Terraform on AWS. ([#2004](https://github.com/pingcap/tidb-operator/pull/2004), [@DanielZhangQD](https://github.com/DanielZhangQD))
- Update TidbConfig for TidbCluster to TiDB v3.1.0 ([#1906](https://github.com/pingcap/tidb-operator/pull/1906), [@Yisaer](https://github.com/Yisaer))
- Allow users to define resources for initContainers in TiDB initializer job. ([#1938](https://github.com/pingcap/tidb-operator/pull/1938), [@tfulcrand](https://github.com/tfulcrand))

> - TLS support for Pump and Drainer ([#1979](https://github.com/pingcap/tidb-operator/pull/1979), [@weekface](https://github.com/weekface))

Reviewer: ```suggestion
- Add TLS support for Pump and Drainer ([#1979](https://github.com/pingcap/tidb-operator/pull/1979), [@weekface](https://github.com/weekface))
```

- 1. Add check to guarantee the NodePort won't be changed if the serviceType of TidbMonitor is NodePort
  2. Add EnvVar sort to avoid the monitor rendering different results from the same TidbMonitor spec.
  3. Fix the problem that Tidbmonitor LoadBalance IP is not used. ([#1962](https://github.com/pingcap/tidb-operator/pull/1962), [@Yisaer](https://github.com/Yisaer))
- Make tidb-initializer support TLS ([#1931](https://github.com/pingcap/tidb-operator/pull/1931), [@weekface](https://github.com/weekface))

> - 1. Fix the problem that Advanced StatefulSet could not work with Webhook

Reviewer: ```suggestion
  1. Fix the problem that Advanced StatefulSet could not work with Webhook
```

Author: this release note contains multiple items, no need to drop `-`.

>   2. Change the Reaction for the Down State Tikv pod during deleting request in webhook from admit to reject. ([#1963](https://github.com/pingcap/tidb-operator/pull/1963), [@Yisaer](https://github.com/Yisaer))

Reviewer: ```suggestion
  2. Change the Reaction for the Down State TiKV pod during deleting request in webhook from admit to reject. ([#1963](https://github.com/pingcap/tidb-operator/pull/1963), [@Yisaer](https://github.com/Yisaer))
```

>   2. Change the Reaction for the Down State TiKV pod during deleting request in webhook from admit to reject. ([#1963](https://github.com/pingcap/tidb-operator/pull/1963), [@Yisaer](https://github.com/Yisaer))

Author: I've updated it to use `webhook`.

>   2. Change the Reaction for the Down State TiKV pod during deleting request in webhook from admit to rejecting. ([#1963](https://github.com/pingcap/tidb-operator/pull/1963), [@Yisaer](https://github.com/Yisaer))

Reviewer: ```suggestion
  2. Change the Reaction for the Down State TiKV pod during deleting request in webhook from admit to reject. ([#1963](https://github.com/pingcap/tidb-operator/pull/1963), [@Yisaer](https://github.com/Yisaer))
```

- Fix some TiKV configuration keys in toml ([#1887](https://github.com/pingcap/tidb-operator/pull/1887), [@aylei](https://github.com/aylei))

> - support using a remote directory as data source for tidb-lightning. ([#1629](https://github.com/pingcap/tidb-operator/pull/1629), [@aylei](https://github.com/aylei))

Reviewer: ```suggestion
- Support using a remote directory as data source for tidb-lightning. ([#1629](https://github.com/pingcap/tidb-operator/pull/1629), [@aylei](https://github.com/aylei))
```

> - Add API document and its generating util ([#1945](https://github.com/pingcap/tidb-operator/pull/1945), [@Yisaer](https://github.com/Yisaer))

Reviewer: Please use the full word of `util`

- Add tikv-importer chart. ([#1910](https://github.com/pingcap/tidb-operator/pull/1910), [@shonge](https://github.com/shonge))
- Fix prometheus scrape config issue while TLS is enabled ([#1919](https://github.com/pingcap/tidb-operator/pull/1919), [@weekface](https://github.com/weekface))
- Enable TLS between TiDB components ([#1870](https://github.com/pingcap/tidb-operator/pull/1870), [@weekface](https://github.com/weekface))

> - Fix the timeout error when .Values.admission.validation.pods is true during tikv upgrading. ([#1875](https://github.com/pingcap/tidb-operator/pull/1875), [@Yisaer](https://github.com/Yisaer))

Reviewer: ```suggestion
- Fix the timeout error when `.Values.admission.validation.pods` is `true` during TiKV upgrading. ([#1875](https://github.com/pingcap/tidb-operator/pull/1875), [@Yisaer](https://github.com/Yisaer))
```

> - Enable TLS For MySQL Clients ([#1878](https://github.com/pingcap/tidb-operator/pull/1878), [@weekface](https://github.com/weekface))

Reviewer: ```suggestion
- Enable TLS for MySQL clients ([#1878](https://github.com/pingcap/tidb-operator/pull/1878), [@weekface](https://github.com/weekface))
```

> - Fix the bug which would cause tidb image property broken ([#1860](https://github.com/pingcap/tidb-operator/pull/1860), [@Yisaer](https://github.com/Yisaer))

Reviewer: ```suggestion
- Fix the bug which would cause broken TiDB image property ([#1860](https://github.com/pingcap/tidb-operator/pull/1860), [@Yisaer](https://github.com/Yisaer))
```

- TidbMonitor would use its namespace for the targetRef if it is not defined. ([#1834](https://github.com/pingcap/tidb-operator/pull/1834), [@Yisaer](https://github.com/Yisaer))
- Support starting tidb-server with `--advertise-address` parameter ([#1859](https://github.com/pingcap/tidb-operator/pull/1859), [@LinuxGit](https://github.com/LinuxGit))
- Backup/Restore: support configuring TiKV GC life time ([#1835](https://github.com/pingcap/tidb-operator/pull/1835), [@LinuxGit](https://github.com/LinuxGit))

> - Support no secret for s3/ceph when using OIDC authentication ([#1817](https://github.com/pingcap/tidb-operator/pull/1817), [@tirsen](https://github.com/tirsen))

Reviewer: ```suggestion
- Support no secret for S3/Ceph when using OIDC authentication ([#1817](https://github.com/pingcap/tidb-operator/pull/1817), [@tirsen](https://github.com/tirsen))
```

> - 1. Change the setting from the previous admission.hookEnabled.pods to the admission.validation.pods.

Reviewer: ```suggestion
  1. Change the setting from the previous `admission.hookEnabled.pods` to the `admission.validation.pods`.
```

  2. Change the setting from the previous admission.hookEnabled.statefulSets to the admission.validation.statefulSets
  3. Change the setting from the previous admission.hookEnabled.validating to the admission.validation.pingcapResources.
  4. Change the setting from the previous admission.hookEnabled.defaulting to the admission.mutation.pingcapResources.
  5. Change the setting from the previous admission.failurePolicy.defaulting to the admission.failurePolicy.mutation.

>   6. Change the setting from the previous admission.failurePolicy.OTHER SETTING to the

Reviewer: ditto. Please use ``` ` ``` to wrap the parameters

  admission.failurePolicy.validation. ([#1832](https://github.com/pingcap/tidb-operator/pull/1832), [@Yisaer](https://github.com/Yisaer))
- Enable TidbCluster defaulting mutation by default which is recommended when admission webhook is used ([#1816](https://github.com/pingcap/tidb-operator/pull/1816), [@Yisaer](https://github.com/Yisaer))

> - Fix a bug that TiKV failed to start while creating cluster using CR with cluster tls enabled ([#1808](https://github.com/pingcap/tidb-operator/pull/1808), [@weekface](https://github.com/weekface))

Reviewer: ```suggestion
- Fix a bug that TiKV fails to start while creating the cluster using CR with cluster TLS enabled ([#1808](https://github.com/pingcap/tidb-operator/pull/1808), [@weekface](https://github.com/weekface))
```

> - Support backup/restore with remote storage with a prefix. ([#1790](https://github.com/pingcap/tidb-operator/pull/1790), [@DanielZhangQD](https://github.com/DanielZhangQD))

Reviewer: ```suggestion
- Support using prefix in remote storage during backup/restore. ([#1790](https://github.com/pingcap/tidb-operator/pull/1790), [@DanielZhangQD](https://github.com/DanielZhangQD))
```
